### PR TITLE
Support enum in attribute arguments

### PR DESCRIPTION
**Pull request by apple-x-co**
## Summary by CodeRabbit

- **New Features**
  - Introduced new annotations `FakeMarker5` and `FakeMarker6` for method annotations.
  - Added PHP 8.1 enum `FakePhp81Enum` with cases `Apple`, `Orange`, and `Grape`.

- **Enhancements**
  - Updated `testVariousMethodSignature` to include additional assertions for new methods.
  - Refactored method signature handling for improved code readability and maintainability.

- **Bug Fixes**
  - Fixed loop condition in tests to cover additional method signatures.

- **Documentation**
  - Updated method annotations to reflect new enum cases and annotations.

### CODE REVIEW

**Comment by apple-x-co** [comment]
@koriym ご確認をよろしくお願いいたします。

**Comment by koriym** [comment]
@apple-x-co 少しリファクタリングするつもりが大きなものになってしまいました><
その代わり、`PHPMD.CyclomaticComplexity`警告抑制を取り除くことができました。

**Comment by koriym** [comment]
@coderabbitai review

File: src/MethodSignatureString.php [hunk 1]
```diff
@@ -53,6 +54,14 @@ public function get(ReflectionMethod $method): string
                 $formattedArgs = [];
 
                 foreach ($argsList as $name => $value) {
+                    if ($value instanceof UnitEnum) {
+                        $formattedValue = preg_replace('/\s+/', ' ', '\\' . var_export($value, true));
+                        $argRepresentation = is_numeric($name) ? $formattedValue : "{$name}: {$formattedValue}";
+                        $formattedArgs[] = $argRepresentation;
+
+                        continue;
+                    }
```
[the remark this answers is not in the document]
Comment (koriym) [reply src/MethodSignatureString.php]: @coderabbitai Is this still valid?

File: src/MethodSignatureString.php [hunk 2]
```diff
@@ -19,87 +20,108 @@
...
 
-        if ($signatureParts) {
-            $signatureParts[] = '    '; // インデント追加
+        if (empty($signatureParts)) {
+            return;
         }
 
-        // アクセス修飾子を取得
+        $signatureParts[] = self::INDENT;
+    }
```
[the remark this answers is not in the document]
Comment (koriym) [reply src/MethodSignatureString.php]: @coderabbitai [1d15b6d](https://github.com/ray-di/Ray.Aop/pull/212/commits/1d15b6dc741342af80ca8b9233af9b9a3d65f1df) enough?

File: src/MethodSignatureString.php [hunk 3]
```diff
@@ -19,87 +20,108 @@
...
+     * @param mixed      $value
+     */
+    private function formatArg($name, $value): string
+    {
+        $formattedValue = $value instanceof UnitEnum ?
+            '\\' . var_export($value, true)
+            : preg_replace('/\s+/', '', var_export($value, true));
 
-        return implode(' ', $signatureParts);
+        return is_numeric($name) ? (string) $formattedValue : "{$name}: {$formattedValue}";
```
[the remark this answers is not in the document]
Comment (koriym) [reply src/MethodSignatureString.php:116]: The original regular expression is sufficient.

File: src/MethodSignatureString.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (koriym) [reply src/MethodSignatureString.php]: `@psalm-pure` is removed.

**Review by NaokiTsuchiya — APPROVED** [review]

File: src/MethodSignatureString.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (koriym) [reply src/MethodSignatureString.php]: @coderabbitai Yes, the problem was that I annotated it when I was not originally qualified to declare it `@psalm-pure`.

**Pull request merged**